parsec/mca/device/cuda/device_cuda_module.c

```diff
@@ -363,25 +363,29 @@ parsec_cuda_module_init( int dev_id, parsec_device_module_t** module )
     cuda_device->minor      = (uint8_t)minor;
     len = asprintf(&gpu_device->super.name, "%s (%d)", szName, dev_id);
     if(-1 == len)
-        gpu_device->super.name = "";
+        gpu_device->super.name = strdup("");
     gpu_device->data_avail_epoch = 0;
 
-    gpu_device->max_exec_streams = PARSEC_MAX_STREAMS;
+    gpu_device->max_exec_streams = 0; // We will increment this as streams are succesfully initialized
     gpu_device->exec_stream =
-        (parsec_gpu_exec_stream_t**)malloc(gpu_device->max_exec_streams * sizeof(parsec_gpu_exec_stream_t*));
+        (parsec_gpu_exec_stream_t**)malloc(PARSEC_MAX_STREAMS * sizeof(parsec_gpu_exec_stream_t*));
```

Comment:
Why are we removing the capability to dynamically define the number of execution streams per device ?

Reply:
This was part of the infrastructure to allow the number of exec_streams to become an MCA param. I would prefer to complete the work than remove it.

Reply:
I updated the PR to add configurable max_streams